### ISSUE TITLE
console readline

### ISSUE THREAD

**Issue by aureooms** [ranger/ranger#669]
```
   READLINE-LIKE BINDINGS IN THE CONSOLE
       ^B, ^F        Move left and right (B for back, F for forward)

       ^P, ^N        Move up and down (P for previous, N for Next)

       ^A, ^E        Move to the start or to the end

       ^D            Delete the current character.

       ^H            Backspace.
```

Console does not allow cut/copy/paste. Would also be nice to have the choice to switch to vi keybindings.

**Comment by mrogalski** [comment]
`Shift+Insert` is the standard readline paste.  Copy from console unfortunately requires mouse: `shift+left mouse button` allows You to select, and paste from the selection buffer.  How do You imagine it should work?
Also, what vi keybindings do You want in the console?  `cmap` or `copycmap` is Your friend here.

**Comment by aureooms** [comment]
Ok, I'll try to play with `cmap` and `copycmap`.
Would it be possible to easily get Normal, Insert and Visual vi modes via `cmap` and `copycmap`.

**Comment by mrogalski** [comment]
No, I don't think You can -- think of the console to be similar to [vim command line](http://vimdoc.sourceforge.net/htmldoc/cmdline.htm).

**Comment by nfnty** [comment]
PRs are welcome.

**Comment by dirdi** [comment]
Why not actually _using_ `readline` via Python's [readline module](https://docs.python.org/3/library/readline.html)? So one can use the bindings one configured in `.inputrc` and it would be backwards compatible to the current implementation for those who did not adapt their `.inputrc`.

**Comment by toonn** [comment]
@dirdi, feel free to PR such a change with thorough evidence that semantics have been maintained. We currently don't have time to replace a part of ranger that works well enough as it is.

**Comment by laktak** [comment]
If anyone is interested in vi mode please help me test #2118 

Here is how to get started: https://github.com/ranger/ranger/wiki/VI-mode-Console
